src/compas_timber/fabrication/btlx.py

```diff
@@ -657,3 +657,53 @@ class EdgePositionType(object):
 
     REFEDGE = "refedge"
     OPPEDGE = "oppedge"
+
+
+class BTLxFromGeometryDefinition(Data):
+    """Container linking a BTLx Process Type and generator function to an input geometry.
+    This allows delaying the actual applying of features to a downstream component.
+
+    Parameters
+    ----------
+    processing : class
+        The BTLx Processing class.
+    geometries : list of :class:`~compas.geometry.Geometry`
+        The geometries to be used as input for the processing.
+    elements : list of :class:`~compas_timber.elements.Element`
+        The elements to be used as input for the processing.
+
+    Attributes
+    processing : class
+        The BTLx Processing class.
+    geometries : list of :class:`~compas.geometry.Geometry`
+        The geometries to be used as input for the processing.
+    elements : list of :class:`~compas_timber.elements.Element`
+        The elements to be used as input for the processing.
+    """
+
+    def __init__(self, processing, geometries, elements=None, **kwargs):
+        super(BTLxFromGeometryDefinition, self).__init__()
+        self.processing = processing
+        self.geometries = geometries if isinstance(geometries, list) else [geometries]
+        if elements:
+            self.elements = elements if isinstance(elements, list) else [elements]
+        else:
+            self.elements = []
+        self.kwargs = kwargs or {}
+
+    @property
+    def __data__(self):
+        return {"processing": self.processing, "geometries": self.geometries, "elements": self.elements, "kwargs": self.kwargs}
+
+    def __repr__(self):
+        return "{}({}, {})".format(BTLxFromGeometryDefinition.__class__.__name__, self.processing, self.geometries)
+
+    def ToString(self):
+        return repr(self)
+
+    def transformed(self, transformation):
```

Comment:
a pythonic pattern tend to follow is having a pair of `transform` and `transformed` methods.
`transform` would do in place:
```
def transform(self, transformation):
    self.geometries = [geo.transformed(transformation) for geo in self.geometries]
```
and `transformed()` would do:
```
def transformed(self, transfomation):
    f_def = self.copy()  # this is a thing because of `Data`
    f_def.transform(transformation)
    return f_def
```